src/Npgsql/NpgsqlParameterCollection.cs

```diff
@@ -476,27 +447,24 @@ IEnumerator<NpgsqlParameter> IEnumerable<NpgsqlParameter>.GetEnumerator()
         /// <inheritdoc />
         public override void AddRange(Array values)
         {
-            foreach (NpgsqlParameter parameter in values)
-                Add(parameter);
+            if (values is null)
+                throw new ArgumentNullException(nameof(values));
+
+            foreach (NpgsqlParameter? parameter in values)
+                Add(parameter ?? throw new ArgumentException("Collection contains a null value", nameof(values)));
```

Comment:
Technically this will leave the collection half-populated, which isn't amazing. It's also the previous behavior, and I wouldn't compromise perf by doing an additional pass just to ensure there aren't any nulls. I wonder what SqlClient does.

Reply:
Let's add `AddRange` which will accept a strongly typed array. SqlClient have that method.